### PR TITLE
[Snyk] Upgrade eslint-config-expensify from 2.0.44 to 2.0.45

### PR DESCRIPTION
**Pull request by MelvinBot**
<p>This PR was automatically created by Snyk using the credentials of a real user.</p><br /><h3>Snyk has created this PR to upgrade eslint-config-expensify from 2.0.44 to 2.0.45.</h3>

As this is a private repository, Snyk-bot does not have access. Therefore, this PR has been created automatically, but appears to have been created by a real user.</br>
:information_source: Keep your dependencies up-to-date. This makes it easier to fix existing vulnerabilities and to more quickly identify and fix newly disclosed vulnerabilities when they affect your project.
<hr/>

- The recommended version is **1 version** ahead of your current version.
- The recommended version was released **21 days ago**, on 2024-04-05.

<details>
<summary><b>Release notes</b></summary>
<br/>
  <details>
    <summary>Package name: <b>eslint-config-expensify</b></summary>
    <ul>
      <li>
        <b>2.0.45</b> - <a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/releases/tag/v2.0.45">2024-04-05</a></br><p>2.0.45</p>
      </li>
      <li>
        <b>2.0.44</b> - <a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/releases/tag/v2.0.44">2024-02-15</a></br><p>2.0.44</p>
      </li>
    </ul>
    from <a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/releases">eslint-config-expensify GitHub release notes</a>
  </details>
</details>

<details>
  <summary><b>Commit messages</b></summary>
  </br>
  <details>
    <summary>Package name: <b>eslint-config-expensify</b></summary>
    <ul>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/530b87dabbb0a21c6893488c3305888bf96a1357">530b87d</a> Merge pull request #90 from Expensify/OSBotify-bump-version-1690aae1-d9db-446b-8e75-d99203ea1b00</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/56d8f94323eee141dc32fecc3c70a80e997f8be2">56d8f94</a> 2.0.45</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/61db348404a7e578f658a0f2e37ac16d79e4b89d">61db348</a> Merge pull request #89 from gedu/edu/avoid_anonymous_fun</li>
      <li><a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/commit/6c8697c8317e009c1c6522d67f327172843b44bf">6c8697c</a> adding avoid anonymous function rule</li>
    </ul>

   <a href="https://snyk.io/redirect/github/Expensify/eslint-config-expensify/compare/72d77f0eae7e66775b98ded3e9c7f7b0a6144dc4...530b87dabbb0a21c6893488c3305888bf96a1357">Compare</a>
  </details>
</details>
<hr/>

**Note:** *You are seeing this because you or someone else with access to this repository has authorized Snyk to open upgrade PRs.*

For more information:  

🧐 [View latest project report](https://app.snyk.io/org/expensify/project/4945ebcf-d145-4524-893a-d9c905019a72?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;upgrade-pr)

🛠 [Adjust upgrade PR settings](https://app.snyk.io/org/expensify/project/4945ebcf-d145-4524-893a-d9c905019a72/settings/integration?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;upgrade-pr)

🔕 [Ignore this dependency or unsubscribe from future upgrade PRs](https://app.snyk.io/org/expensify/project/4945ebcf-d145-4524-893a-d9c905019a72/settings/integration?pkg&#x3D;eslint-config-expensify&amp;utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;upgrade-pr#auto-dep-upgrades)

### CODE REVIEW

**Comment by lakchote** [comment]
@MariaHCD you are assigned because I had to do some changes to make the version bump comply with linting.

File: lib/components/form/element/combobox.js [hunk 1]
```diff
@@ -618,15 +627,16 @@ class Combobox extends React.Component {
      * @param {number} newFocusedIndex
      */
     switchFocusedIndex(oldFocusedIndex, newFocusedIndex) {
-        this.setState((state) => {
+        const setValueState = (state) => {
```
Comment (MariaHCD) [inline lib/components/form/element/combobox.js:630]:
```suggestion
        const setFocusedState = (state) => {
```

File: lib/components/form/element/dropdown.js [hunk 2]
```diff
@@ -49,34 +49,41 @@ const defaultProps = {
...
-            </ul>
+            <DropDownItem
+                key={_.uniqueId()}
+                bs4={this.props.bs4}
+                value={option.value}
+                text={option.text}
+                disabled={option.disabled}
+                focused={option.focused}
+                isSelected={option.isSelected}
+                isSelectable={option.isSelectable}
```
Comment (MariaHCD) [inline lib/components/form/element/dropdown.js:75]: Just curious, how did you verify these were all the props on `option`?

**Review by MariaHCD — COMMENTED** [review]
Looks good overall, just some minor suggestions / questions.

Additionally, this touches quite a bit of logic, can we test this against the App to make sure things are still working as expected?

File: lib/components/form/element/combobox.js [hunk 3]
```diff
@@ -543,20 +545,25 @@ class Combobox extends React.Component {
         const optionMatchingVal = _.findWhere(this.options, {value: val});
         const currentText = get(optionMatchingVal, 'text', '');
 
-        this.initialValue = val;
-        this.setState((state) => ({
+        const deselectLogic = (o) => {
```
[the remark this answers is not in the document]
Comment (lakchote) [reply lib/components/form/element/combobox.js]: Yes, I've modified code. Let me know if this sounds better.

File: lib/ExpensiMark.js [hunk 4]
```diff
@@ -176,7 +177,7 @@ export default class ExpensiMark {
             {
                 name: 'reportMentions',
 
-                regex: /(?<![^ \n*~_])(#[\p{Ll}0-9-]{1,80})/gmiu,
+                regex: /(?<![^ \n*~_])(#[\p{Ll}0-9-]{1,80})/gimu,
```
[the remark this answers is not in the document]
Comment (lakchote) [reply lib/ExpensiMark.js:180]: It was a typo on my part, it's fixed.

**Comment by lakchote** [comment]
> Additionally, this touches quite a bit of logic, can we test this against the App to make sure things are still working as expected?

I do agree. 

@blazejkustra hi! You've helped me in my last PR in `expensify-common` which had the same kind of big changes/refactoring. Would you be up to test everything works correctly again? I'd greatly appreciate it 🙏

**Comment by blazejkustra** [comment]
@lakchote Sure I'll test it tomorrow morning, from what I see only `str` and `ExpensiMark` are used in E/App, but changes around `str` are marginal. I'll focus on testing `ExpensiMark` functionality in the E/App. Code looks good to me 😄

**Comment by blazejkustra** [comment]
@lakchote Hey! Seems that everything works fine! Tested on all platforms:

<summary>Show</summary>

https://github.com/Expensify/expensify-common/assets/46095609/a4344e47-fc39-49c1-bc34-b579c5cb3d4a

https://github.com/Expensify/expensify-common/assets/46095609/00233c1d-cc1f-4a6c-bfae-24b25b6d1fff

https://github.com/Expensify/expensify-common/assets/46095609/e5bc72e0-a95e-4996-a2eb-06b8d5b0fb1f

[android.webm](https://github.com/Expensify/expensify-common/assets/46095609/0135e5ee-2087-4072-9bdb-8fc7c3f5f5bd)

</details>

**Comment by lakchote** [comment]
> @lakchote Hey! Seems that everything works fine! Tested on all platforms:

Thank you so much @blazejkustra! 

@MariaHCD could you please give this a final review?

File: lib/components/form/element/combobox.js [hunk 5]
```diff
@@ -835,11 +853,13 @@ class Combobox extends React.Component {
         }
         matches = Array.from(matches);
 
-        const options = _(matches).map((option) => ({
+        const filterOption = (option) => ({
```
Comment (MariaHCD) [inline lib/components/form/element/combobox.js:856]:
```suggestion
        const formatOption = (option) => ({
```

We're not filtering options here, we just formatting them, right?

**Review by MariaHCD — COMMENTED** [review]
Just a few small suggestions

File: lib/ExpensiMark.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (lakchote) [reply lib/ExpensiMark.js]: Yes, done [here](https://github.com/Expensify/expensify-common/pull/689/commits/0520e5adb011c2083fbee08b78f5d300917f1113)

File: lib/components/form/element/combobox.js (hunk 5 above)
Comment (lakchote) [reply lib/components/form/element/combobox.js]: Yes, I've modified the naming to reflect that.

**Comment by lakchote** [comment]
Handled your comments @MariaHCD, thank you!

**Review by MariaHCD — APPROVED** [review]

**Pull request merged**